docs/regression_test_api.rst

```diff
@@ -87,6 +87,40 @@ For instance, continuing with the example above, one could override the :func:`_
        This only has an effect if used with ``inherit_params=True``.
 
 
+.. py:function:: reframe.core.pipeline.RegressionTest.var(name, *types, value=None, field=None)
+
+   Inserts a new regression test variable.
+   If the ``value`` argument is not provided, the variable is considered *declared* but not *defined*.
+   Thus, a variable may not be declared more than once. However, it is possible to alter a variable's value after it was declared by using the :func:`set_var` and :func:`require_var` directives.
+   Note that a variable must be defined before is referenced in the regression test. Otherwise, an :py:exc:`AttributeError` will be raised.
+
+   :param name: the variable name.
+   :param types: the supported types for the variable.
+   :param value: the default value assigned to the variable.
+   :param field: the field validator to be used for this variable.
+        If no field argument is provided, it defaults to
+        :class:`reframe.core.fields.TypedField`.
```

Comment:
This interface is not exposed to the users, so we will need to do so. I see it relevant for test library developers, but we should expose only a subset of the fields.

Reply:
Agreed for both.